### PR TITLE
Provide jobs endpoint to SDK 

### PR DESCRIPTION
**Pull request by javiermtorres**
## What's changing

This PR provides a new SDK class to handle jobs, following https://github.com/mozilla-ai/lumigator/pull/285.

Also, mock tests and integration tests (when the mentioned PR is merged in) will also be provided.

## I already...

- [x] added some tests for any new functionality;
- [ ] updated the documentation.

### CODE REVIEW

**Comment by javiermtorres** [comment]
The `wait_for_job` method is intended to be run in an async context. Since Ray is used directly to check the jobs, changes are needed to provide the Ray jobs URL (and possibly depend on the `ray[default]` package as indicated [here](https://docs.ray.io/en/latest/cluster/running-applications/job-submission/sdk.html), although this seems really overkill).

File: lumigator/python/mzai/sdk/sdk/experiments.py [hunk 1]
```diff
@@ -1,77 +0,0 @@
-from http import HTTPMethod, HTTPStatus
-from json import dumps
-from uuid import UUID
-
-from schemas.experiments import (
-    ExperimentCreate,
```
Comment (ividal) [inline lumigator/python/mzai/sdk/sdk/experiments.py:6]: I wouldn't get rid of experiments just yet.
On the backend side it's a dummy right now, but the intention is to use to to trigger a very specific workflow/combination of jobs, that is, the 2 that used to be the "old" experiments:

One of:
* evaluation job, assuming there are predictions and labels (aka ground_truth)
or
* inference (ground_truth generation) job, evaluation job (assuming we only have predictions)

In the future, it would make most sense to generate the predictions as well.

None of this is in main, though, and experiemnts right now just calls jobs, so at least the boilerplate with a get for a list of experiemnts is worth keeping (I think).

Comment (javiermtorres) [reply lumigator/python/mzai/sdk/sdk/experiments.py:6]: Ok. I'll keep the experiments endpoints just in case.

File: lumigator/python/mzai/sdk/sdk/jobs.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            id (UUID): the id of the job to wait for
+            retries (int):
+            poll_wait (int):
+        Returns:
+            JobResponse: the most recently job information for the id, when the job has finished
+        """
+        for _ in range(1, retries):
+            # http://localhost:8265/api/jobs/f311fa44-025a-4703-b8ba-7e0b1001b484
+            # response = self.client.get_response(f"{self.JOBS_ROUTE}/{id}")
+            response = requests.get(f"http://localhost:8265/api/jobs/{id}")
```
[the remark this answers is not in the document]
Comment (javiermtorres) [reply lumigator/python/mzai/sdk/sdk/jobs.py:70]: Heavens no :D Neither the Ray hostname. But this is very preliminary: some people think the SDK should not even deal with the jobs API, but I believe it's good that we codify some common use patterns.

File: lumigator/python/mzai/sdk/tests/conftest.py [hunk 3]
```diff
@@ -41,14 +41,21 @@ def mock_vendor_data() -> str:
 def resources_dir() -> Path:
     return Path(__file__).parent / "data"
 
+
```
[the remark this answers is not in the document]
Comment (javiermtorres) [reply lumigator/python/mzai/sdk/tests/conftest.py:44]: @peteski22 any comments on this?

File: lumigator/python/mzai/sdk/sdk/jobs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ividal) [reply lumigator/python/mzai/sdk/sdk/jobs.py:70]: I gathered at much, I didn't expect a number staring at me from your PR :D 

Re: jobs, I think it does make sense for the SDK to handle it. An experiment could have a strict order of jobs that the sdk doesn't need to know about... but we do want to handle stand-alone inference and I'm not sure we should add an endpoint outside jobs for that.

Shortly after, we'll add more functionality: different forms of dataset analysis could be 1..n jobs (either ours or contributed): isn't it helpful if the SDK can launch them? What about explanations of the models? --> Jobs 🧰 

TL;DR I think a jobs handle in the SDK is still very useful.

File: lumigator/python/mzai/sdk/tests/conftest.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (veekaybee) [reply lumigator/python/mzai/sdk/tests/conftest.py:44]: yep, this should be a fixture so we can inject into other tests per session

File: lumigator/python/mzai/sdk/sdk/jobs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (javiermtorres) [reply lumigator/python/mzai/sdk/sdk/jobs.py:70]: I'll include the call using the jobs sdk, but I'll provide a param to the lumigator client with a default value of `127.0.0.1:8265` to set the Ray jobs API connection. We can then remove it easily later on if we need to.

File: lumigator/python/mzai/sdk/tests/conftest.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (javiermtorres) [reply lumigator/python/mzai/sdk/tests/conftest.py:44]: Done

File: lumigator/python/mzai/sdk/sdk/jobs.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+from sdk.client import ApiClient
+
+
+class Jobs:
+    JOBS_ROUTE = "jobs"
+
+    def __init__(self, c: ApiClient):
+        self.client = c
+
+    def get_jobs(self) -> ListingResponse[JobResponse]:
```
[the remark this answers is not in the document]
Comment (ividal) [reply lumigator/python/mzai/sdk/sdk/jobs.py:29]: ⬆️  @javiermtorres We're tracking it as LUMI-194.

File: lumigator/python/mzai/sdk/sdk/jobs.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+            id (UUID): the id of the job to wait for
+            retries (int):
+            poll_wait (int):
+        Returns:
+            JobResponse: the most recently job information for the id, when the job has finished
+        """
+        for _ in range(1, retries):
+            # http://localhost:8265/api/jobs/f311fa44-025a-4703-b8ba-7e0b1001b484
+            response = self.client.get_ray_job_response(f'{id}')
+            # response = requests.get(f"http://localhost:8265/api/jobs/{id}")
```
Comment (ividal) [inline lumigator/python/mzai/sdk/sdk/jobs.py:70]: Nit and non-blocking: I would remove these commented lines 🤫

**Review by ividal — APPROVED** [review]

File: lumigator/python/mzai/sdk/sdk/jobs.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (javiermtorres) [reply lumigator/python/mzai/sdk/sdk/jobs.py:29]: Ok, but it would be out of the scope of this PR, right?

File: lumigator/python/mzai/sdk/sdk/client.py [hunk 6]
```diff
@@ -96,3 +98,37 @@ def get_response(
...
+        method: HTTPMethod = HTTPMethod.GET,
+        data=None,
+        files=None,
+        json_data=None,
+        verbose: bool = True,
+    ) -> requests.Response:
+        """Makes a request to the specified path and attempts to return the response.
+        Raises an exception for any error other than 404 - NOT FOUND.
+        """
+        path = f"{self._ray_url.rstrip('/')}/{api_path.lstrip('/')}"
```
[the remark this answers is not in the document]
Comment (javiermtorres) [reply lumigator/python/mzai/sdk/sdk/client.py:114]: My opinion is, let's leave it for the moment since it's useful to wait for jobs to stop, and later on we can choose.

File: lumigator/python/mzai/sdk/sdk/jobs.py (hunk 5 above)
Comment (javiermtorres) [reply lumigator/python/mzai/sdk/sdk/jobs.py:70]: Done

File: lumigator/python/mzai/sdk/sdk/client.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (ividal) [reply lumigator/python/mzai/sdk/sdk/client.py:114]: Opened LUMI-195 to track this and we can discuss it with Peteski 😉

[the remark this answers is not in the document]
Comment (chainlink) [reply lumigator/python/mzai/sdk/sdk/client.py:114]: Agree with not exposing Ray via the API

[the remark this answers is not in the document]
Comment (veekaybee) [reply lumigator/python/mzai/sdk/sdk/client.py:114]: Could you share the branch link when it's available for discussion? I'd prefer to not expose app internals in an SDK, and additionally there are better ways of handling pathing in Python (i.e. with Pathlib)

[the remark this answers is not in the document]
Comment (ividal) [reply lumigator/python/mzai/sdk/sdk/client.py:114]: @javiermtorres Let´s talk about the endpoint that you can use and LUMI-195 .

[the remark this answers is not in the document]
Comment (javiermtorres) [reply lumigator/python/mzai/sdk/sdk/client.py:114]: Please refer to https://github.com/mozilla-ai/lumigator/pull/313, I'll include any necessary changes there.
